File: 2725-mice-and-cheese/2725-mice-and-cheese.java
class Solution {
    public int miceAndCheese(int[] reward1, int[] reward2, int k) {

        PriorityQueue<Integer>r1=new PriorityQueue<>(Collections.reverseOrder());
       int sum=0;

        for(int i=0;i<reward1.length;i++)
        {
            r1.offer(reward1[i]-reward2[i]);
            sum+=reward2[i];

           
        }
        
        
        for(int i=0;i<k;i++)
        {
            sum+=r1.poll();
        }

      

        
        return sum;
    }
}

File: 2725-mice-and-cheese/2725-mice-and-cheese.py
import heapq
class Solution:
    def miceAndCheese(self, reward1: List[int], reward2: List[int], k: int) -> int:
        r1=[]
        total_sum=sum(reward2)

        for i in range(len(reward1)):
            heapq.heappush(r1,-(reward1[i]-reward2[i]))

        for i in range(k):
            total_sum+=-heapq.heappop(r1)

        return total_sum
        